### PR TITLE
fix: sync changes added to mobile branch to main

### PR DESCRIPTION
**Pull request by wodoro**
fixes the test to pass
tightens the uri checks 

## Summary by CodeRabbit

* **Bug Fixes**
  * Tightened HTTP request validation to more consistently reject unsafe or malformed URIs, including checks on the query component to prevent traversal and encoding edge cases. This may reduce acceptance of previously borderline requests.
* **Tests**
  * Expanded and refined validation tests to cover additional URI/query edge cases and standardized assertion style for consistency.

### CODE REVIEW

File: http-api/src/test/java/bisq/http_api/validator/RequestValidatorIsValidPathTest.java [hunk 1]
```diff
@@ -30,7 +29,7 @@ void testEmptyPath() {
     @Test
     void testDotDotInPath() {
         RequestValidator validator = new RequestValidator(List.of(), List.of());
-        assertFalse(validator.isValidPath("/foo/../bar"));
+        assertFalse(validator.isValidPath("../foo/../bar"));
```
[the remark this answers is not in the document]
Comment (wodoro) [reply http-api/src/test/java/bisq/http_api/validator/RequestValidatorIsValidPathTest.java]: this is intentional

**Review by HenrikJannsen — APPROVED** [review]
utACK
